### PR TITLE
Change welcome message for players with their chat turned off.

### PR DESCRIPTION
**Pull request by ConorGriffin37**

### CODE REVIEW

File: CoreScriptsRoot/Modules/Server/ServerChat/ChatServiceRunner.lua [hunk 1]
```diff
@@ -294,6 +297,21 @@ local systemChannel = ChatService:AddChannel("System")
...
+	if RunService:IsStudio() then
+		return
+	end
+	local player = speaker:GetPlayer()
+	if player then
+		local success, canChat = pcall(function()
+			return Chat:CanUserChatAsync(player.UserId)
+		end)
+		if success and not canChat then
+			return "Enable in-game chat in your privacy settings to be able to chat."
```
Comment (DarraghGriffin) [inline CoreScriptsRoot/Modules/Server/ServerChat/ChatServiceRunner.lua]: This seems like a call for players to change their privacy settings? Do we want to do this?

Comment (SolarCrane) [inline CoreScriptsRoot/Modules/Server/ServerChat/ChatServiceRunner.lua]: Hi Darragh, I appreciate that you are code reviewing your own pull request. For strings presented to end-users we should get approval from the PM corresponding to this feature.

**Comment by kessanro** [comment]
FUCK OFF ROBLOX YOU ARE CRINGE

**Comment by Mdstumpfy123** [comment]
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE
FUCK OFF ROBLOX YOU ARE CRINGE

File: CoreScriptsRoot/Modules/Server/ClientChat/ChatBar.lua [hunk 2]
```diff
@@ -5,6 +5,14 @@
 local module = {}
 
 local UserInputService = game:GetService("UserInputService")
+local RunService = game:GetService("RunService")
+local Players = game:GetService("Players")
+local LocalPlayer = Players.LocalPlayer
+
+while not LocalPlayer do
+	Players.ChildAdded:wait()
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/Server/ClientChat/ChatBar.lua]: why not PlayersAdded?

File: CoreScriptsRoot/Modules/Server/ClientChat/ChatBar.lua [hunk 3]
```diff
@@ -113,9 +121,31 @@ function methods:CreateGuiObjects(targetParent)
...
+			self.TextLabel.Text = "Sign up to chat in game."
+		end
+		self:CalculateSize()
+	end
+	if self.TextBox then
+		self.TextBox.Active = false
+		local textboxfocusedConnection = self.TextBox.Focused:connect(function()
+			self.TextBox:ReleaseFocus()
+		end)
+		table.insert(self.Connections, textboxfocusedConnection)
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/Server/ClientChat/ChatBar.lua]: whats the point in storing this connection? do all the connections get destroyed when something in here is destroyed?

File: CoreScriptsRoot/Modules/Server/ClientChat/ChatBar.lua [hunk 4]
```diff
@@ -129,14 +159,23 @@ function methods:SetUpTextBoxEvents(TextBox, TextLabel, MessageModeTextButton)
...
+			if (self:IsFocused() and TextBox.Text == "") then
 				self:SetChannelTarget(ChatSettings.GeneralChannelName)
 			end
 		end
 	end)
 	table.insert(self.Connections, inputBeganConnection)
 
+
+	local calculatingSizeLock = false
 	local textboxChangedConnection = TextBox.Changed:connect(function(prop)
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/Server/ClientChat/ChatBar.lua]: do we do anything with this connection?

File: CoreScriptsRoot/Modules/Server/ClientChat/ChatBar.lua [hunk 5]
```diff
@@ -240,11 +287,19 @@ function methods:GetEnabled()
 end
 
 function methods:SetEnabled(enabled)
-	self.GuiObject.Visible = enabled
+	if not self.UserHasChatOff then
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/Server/ClientChat/ChatBar.lua]: I'd prefer if we didn't use a not here, can we switch the statement around?

File: CoreScriptsRoot/Modules/Server/ClientChat/ChatBar.lua [hunk 6]
```diff
@@ -503,6 +568,17 @@ function module.new(CommandProcessor, ChatWindow)
 		end
 	end)
 
+	coroutine.wrap(function()
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/Server/ClientChat/ChatBar.lua]: y u no spawn

Comment (ConorGriffin37) [inline CoreScriptsRoot/Modules/Server/ClientChat/ChatBar.lua]: Spawn starts on the next frame, coroutine.wrap starts this frame.

**Pull request merged**